### PR TITLE
0022121: Failed test: Filter "Staff List" View

### PR DESCRIPTION
**Pull request by bseglias**
https://www.ilias.de/mantis/view.php?id=22121

### CODE REVIEW

File: Services/Repository/classes/class.ilRepositorySelectorExplorerGUI.php [hunk 1]
```diff
@@ -377,8 +377,7 @@ function isNodeClickable($a_node)
 			// return false; #14354
 		}
 
-		if (!$ilAccess->checkAccess("read", "", $a_node["child"]))
-		{
+		if(!$ilAccess->hasUserRBACorAnyPositionAccess("read",$a_node["child"])) {
```
Comment (chfsx) [inline Services/Repository/classes/class.ilRepositorySelectorExplorerGUI.php:380]: @smeyer-ilias Is that ok for you?

File: Services/AccessControl/classes/class.ilAccess.php [hunk 2]
```diff
@@ -899,4 +899,18 @@ public function filterUserIdsByPositionOfUser($user_id, $pos_perm, $ref_id, arra
 	public function filterUserIdsByRbacOrPositionOfCurrentUser($rbac_perm, $pos_perm, $ref_id, array $user_ids) {
 		return $this->ilOrgUnitPositionAccess->filterUserIdsByRbacOrPositionOfCurrentUser($rbac_perm, $pos_perm, $ref_id,  $user_ids);
 	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function hasCurrentUserAnyPositionAccess($ref_id) {
```
Comment (chfsx) [inline Services/AccessControl/classes/class.ilAccess.php:906]: @smeyer-ilias there are several new method in the position-related interfaces which we had to implement in ilAccess, too. OK for you?

**Review by chfsx — APPROVED** [review]
LGTM in general, I'll wait for feedback from @smeyer-ilias before merging.

**Comment by chfsx** [comment]
to JF: there is a small interface-change in ilAccessHandler

**Comment by jf** [comment]
what? who? why? how? noo way!

**Comment by chfsx** [comment]
sorry @jf! :D wanted to mention our "JourFixe"...

**Comment by chfsx** [comment]
@smeyer-ilias would be great if you could have a look at this today, there are several testcases depending on it. thank you very much!

**Comment by smeyer-ilias** [comment]
Sorry, Fabian. Haven't seen it until now.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 22 JAN 2018 : PR highly appreciated. Stefan will merge PR to trunk and 5.3.

**Pull request merged**